lib/matplotlib/axes/_base.py

```diff
@@ -3203,6 +3203,12 @@ def set_xticks(self, ticks, minor=False):
         """
         ret = self.xaxis.set_ticks(ticks, minor=minor)
         self.stale = True
+
+        g = self.get_shared_x_axes()
+        for ax in g.get_siblings(self):
+            ax.xaxis.set_ticks(ticks, minor=minor)
```

Comment:
Given how sharing works, I don't think you need all these "set_ticks" lines.  I suspect that simply setting 'ax.stale = True' is enough.  Would you try that, please?

Reply:
I just tried it (commenting line 3209). Unfortunately doing that fails the automated test I created, so it wouldn't fix the issue.

Reply:
Thank you.  Still, I'm pretty sure that calling set_ticks is not the right thing to do here.  It is breaking part of the sharing connection by giving the axes different instances of the Locators. Instead, it looks like what is needed is synchronization of the view_interval after it was changed in the call to set_ticks.  Logically, it seems like sharing should include the viewLim directly, but it doesn't, so when the viewLim is changed, the other axes need to be notified.